promql/engine.go

```diff
@@ -3313,8 +3320,45 @@ func setOffsetForAtModifier(evalTime int64, expr parser.Expr) {
 	})
 }
 
+func detectHistogramStatsDecoding(expr parser.Expr) {
```

Comment:
Let's have a doc comment here explaining what this does.

Reply:
Added a comment.